### DISCUSSION TITLE
Questions about try catch

### DISCUSSION THREAD

**Discussion by luozongle** [apache/rocketmq#6667] [Q&A]
Why does this code use two tries? Is there any optimization in it compared to using one try catch finally

<img width="920" alt="image" src="https://user-images.githubusercontent.com/35722577/235298560-49000a6b-d9a9-49ca-a28d-a3d2c836a412.png">

**Comment by RongtongJin** [comment]
> Why does this code use two tries? Is there any optimization in it compared to using one try catch finally
> 
> <img alt="image" width="920" src="https://user-images.githubusercontent.com/35722577/235298560-49000a6b-d9a9-49ca-a28d-a3d2c836a412.png">

Of course, welcome to optimize this part of the code
